core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -901,6 +900,67 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleCreateTopicRequest(request: RequestChannel.Request) {
+    val createTopicRequest = request.body.asInstanceOf[CreateTopicRequest]
+
+    def handleUnauthorizedRequest = {
+      createTopicRequest.topics.asScala.map { case (topic, _) =>
+        (topic, Errors.CLUSTER_AUTHORIZATION_FAILED)
+      }
+    }
+
+    def handleAuthorizedRequest = {
+      val brokers = metadataCache.getAliveBrokers.map(_.id)
+      createTopicRequest.topics.asScala.map { case (topic, arguments) =>
```

Comment:
Since the list of topics stored in a Map in the Request object we are guaranteed to get only one instance of a given topic.